gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/util/CodeableConceptUtilTest.java

```diff
@@ -0,0 +1,181 @@
+        CD c1 = new CD();
+        assertFalse(CodeableConceptUtil.compareCodeableConcepts(c1, null));
+    }
+
+    @Test
+    public void compareCodeableConceptsWithOneTransaction() {
+
+        CD c1 = new CD();
+        c1.setCode("Code1");
+        c1.setCodeSystem("CodeSystem1");
+        c1.setDisplayName("DisplayName1");
+        c1.setOriginalText("OriginalText");
+
+        CD translationCD1 = new CD();
+        translationCD1.setCode("161586000");
+        translationCD1.setCodeSystem("2.16.840.1.113883.2.1.3.2.4.15");
+        translationCD1.setDisplayName("H/O: injury");
+        c1.getTranslation().add(translationCD1);
+
+        CD c2 = new CD();
+        c2.setCode("Code1");
+        c2.setCodeSystem("CodeSystem1");
+        c2.setDisplayName("DisplayName1");
+        c2.setOriginalText("OriginalText");
+
+        CD translationCD2 = new CD();
+        translationCD2.setCode("161586000");
+        translationCD2.setCodeSystem("2.16.840.1.113883.2.1.3.2.4.15");
+        translationCD2.setDisplayName("H/O: injury");
+        c2.getTranslation().add(translationCD2);
+
+        assertTrue(CodeableConceptUtil.compareCodeableConcepts(c1, c2));
+    }
+
+    @Test
+    public void compareCodeableConceptsWithOneQualifierPair() {
+
+        CD c1 = new CD();
+        c1.setCode("Code1");
+        c1.setCodeSystem("CodeSystem1");
+        c1.setDisplayName("DisplayName1");
+        c1.setOriginalText("OriginalText");
+
+        CR cr1 = new CR();
+        cr1.setCode("QualifierCode");
+        cr1.setCodeSystem("QualifierCodeSystemCode");
+        cr1.setDisplayName("QualifierDisplayName");
+        c1.getQualifier().add(cr1);
+
+        CD c2 = new CD();
+        c2.setCode("Code1");
+        c2.setCodeSystem("CodeSystem1");
+        c2.setDisplayName("DisplayName1");
+        c2.setOriginalText("OriginalText");
+
+        CR cr2 = new CR();
+        cr2.setCode("QualifierCode");
+        cr2.setCodeSystem("QualifierCodeSystemCode");
+        cr2.setDisplayName("QualifierDisplayName");
+        c2.getQualifier().add(cr2);
+
+        assertTrue(CodeableConceptUtil.compareCodeableConcepts(c1, c2));
+    }
+
+    @Test
+    public void compareCodeableConceptsWithTwoTransactionsPairsNotIdenticalTranslations() {
+
+        CD c1 = new CD();
+        c1.setCode("Code1");
+        c1.setCodeSystem("CodeSystem1");
+        c1.setDisplayName("DisplayName1");
+        c1.setOriginalText("OriginalText");
+
+        CD translationCD1 = new CD();
+        translationCD1.setCode("161586000");
+        translationCD1.setCodeSystem("2.16.840.1.113883.2.1.3.2.4.15");
+        translationCD1.setDisplayName("H/O: injury");
+        c1.getTranslation().add(translationCD1);
+
+        CD translationCD2 = new CD();
+        translationCD2.setCode("161586007");
+        translationCD2.setCodeSystem("2.16.840.1.113883.2.1.3.2.4.17");
+        translationCD2.setDisplayName("H/O: injury7");
+        c1.getTranslation().add(translationCD2);
```

Comment:
Did you intentionally not add the translations to the Codable Concept. If so why create them in the first place?

Reply:
Translation expects to receive items of type CD which is different type to CodeableConcept. Symantically they are similar  in the way how we use the fields(Code, CodeSystem and DisplayName fields) but they are still different objects with different names -> CD vs. CodeableConcept